Firmware/Main/CAN.cpp

```diff
@@ -10,43 +10,57 @@
 /* CAN bus handle */
 FlexCAN_T4<CAN2, RX_SIZE_256, TX_SIZE_16> CAN_bus;
 
-/* CAN struct for reading/writing */
-CAN_message_t CAN_msg;
+/* Handle for a queue storing received CAN messages */
+QueueHandle_t messageQueue;
+
+/* Stores the latest message pulled from the queue */
+queue_message received_msg;
 
 void initCAN() {
   CAN_bus.begin();
   CAN_bus.setBaudRate(250000);
+  CAN_bus.setMaxMB(16);
+  CAN_bus.enableFIFO();
 }
 
-void decipherEVCCStats(CAN_message_t msg, ChargeControllerStats *evcc_stats) {
+void addToQueue(const CAN_message_t &msg) {
+  queue_message q_msg;
+  q_msg.id = msg.id;
+  for (int i = 0; i < msg.len; i++) {
+    q_msg.buf[i] = msg.buf[i];
+  }
+  xQueueSendToBackFromISR(messageQueue, &q_msg, NULL);
```

Comment:
Any reason not to use `xQueueSendFromISR()`?

Reply:
The FreeRTOS docs said "xQueueSendFromISR() was the original version and it is now recommended to use xQueueSendToBackFromISR() in its place" but it didn't really say why...........maybe send to front is new?